### ISSUE TITLE
403 Error for YouTube links

### ISSUE THREAD

**Issue by newpen** [leedo/noembed#115]
For example:
https://noembed.com/embed?callback=%3Cesi%3Ainclude%20src%3D%22%2Fesi%2Fjsonp-callback%22%2F%3E&url=https://www.youtube.com/watch?v=G68Q4lCM5pQ

**Comment by komlenic** [comment]
Same issue. All youtube links return 403 and cannot be embedded.

**Comment by momoko8443** [comment]
same issue!

**Comment by gummyfrog** [comment]
ditto. doesn't look like an issue with the provider template– maybe a youtube deprecation? still works fine for most older videos.

**Comment by aariacarterweir** [comment]
Some youtube urls are still working. Cached?

**Comment by lpsBetty** [comment]
![image](https://user-images.githubusercontent.com/244475/104165553-d3c81780-53f9-11eb-85b2-d12aab78e0ee.png)

got the same error when calling the URL in the browser directly, when trying to fetch via JSONP, i got an 200, but the embed does not work anymore... :/ 

I think youtube changed something!

**Comment by aariacarterweir** [comment]
This was a bit of a wig out for me so I quickly pulled together this docker image based on iframely, who kindly release their software as open source. They actually have a Dockerfile in their repo that is pretty much ready to go but theirs is based on node 12 which doesn't work so great with http 2. That and I wanted something a bit more portable that cloned their repo into the actual docker image.

https://hub.docker.com/repository/docker/aariacarterweir/docker-iframely

Won't be a solution for everyone but works pretty well for me. Only thing I'd say is you reallllly gotta run it with redis or memcached so you dont hit your API limits.

**Comment by nicolochiellini** [comment]
Youtube like Vimeo now ask an API key

**Comment by gummyfrog** [comment]
Fixed on local install by installing Mozilla::CA and explicitly using HTTPS for YouTube URLs. 

Here is the full error hiding behind 403 Forbidden:
`{
  "error": {
    "code": 403,
    "message": "SSL is required to perform this operation.",
    "status": "PERMISSION_DENIED"
  }
}`

**Comment by gummyfrog** [comment]
Seems like &scheme isn't cutting it anymore. 

`http://www.youtube.com/oembed/?url=http%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DcVRyoUJyxIE&scheme=https` (generated by Providers/YouTube.pm) still uses HTTP, giving us 403.

**Comment by komlenic** [comment]
I know this github project is essentially abandoned, but does anyone know if [noembed.com](https://noembed.com) is ever updated? A lot of sites are relying on that service and have broken youtube embeds right now.

**Comment by flipmcf** [comment]
look at #114 - we might have to follow what @tomgross has done.

**Comment by eduardopiasson-br** [comment]
A temporary solution can be this, incorporates the video, the rest depends on your application.

I replaced the "https://noembed.com/embed?nowrap=on" with "https://www.youtube.com/oembed?" in the proxy configuration of the trumbowyg api.

the others that use noembed I believe it is almost the same situation.

![Captura de tela 2021-01-13 103603](https://user-images.githubusercontent.com/61803287/104459497-91364480-558b-11eb-8a81-6c7445839dce.png)

Hope this helps.

**Comment by lpsBetty** [comment]
@eduardopiasson-br the problem with this is, that i would need JSONP and youtube doesn't support that :/

**Comment by eduardopiasson-br** [comment]
@lpsBetty Even if you pass the format to the end of the link?

Exemple: https://www.youtube.com/oembed?url=https://www.youtube.com/watch?v=IJ5DWUu-MlE&format=json

**Comment by lpsBetty** [comment]
@eduardopiasson-br yes if you click the link you see that you do not have a callback function in the code - i would need to define a callback function for CKEDITOR...

**Comment by theuros** [comment]
I noticed that for youtube videos which gives 403 error you can use embed url to make it work. Youtube link is like https://www.youtube.com/watch?v=....  and if you click "share" and copy given url https://youtu.be/... then embed works almost fine. Embeded video height is too small

**Comment by dlopez-akalam** [comment]
We where facing the same problem, but It looks like youtube embed codes are working again. The examples on this issue and in our site are working back.

**Comment by vindhya124** [comment]
@eduardopiasson-br 

I am getting not found error when i am trying to play the video because of below url.
https://www.youtube.com/oembed?url=https://www.youtube.com/embed/Ke90Tje7VS0

Can you help me how to fix it.
